src/k8s/pkg/k8sd/features/calico/internal.go

```diff
@@ -0,0 +1,176 @@
+		}
+		autodetectionV4Key = "firstFound"
+		autodetectionV4Value = v == "true"
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4Kubernetes); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Kubernetes)
+		}
+		autodetectionV4Key = "kubernetes"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4Interface); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4Interface)
+		}
+		autodetectionV4Key = "interface"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4SkipInterface); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4SkipInterface)
+		}
+		autodetectionV4Key = "skipInterface"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4CanReach); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4CanReach)
+		}
+		autodetectionV4Key = "canReach"
+		autodetectionV4Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV4CIDRs); ok {
+		if autodetectionV4Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v4 annotations found: %s", annotationAutodetectionV4CIDRs)
+		}
+		autodetectionV4Key = "cidrs"
+		autodetectionV4Value = strings.Split(v, ",")
+	}
+
+	// If any annotation is set, pass the map to the config otherwise it's left nil
+	if autodetectionV4Key != "" {
+		config.autodetectionV4 = map[string]any{
+			autodetectionV4Key: autodetectionV4Value,
+		}
+	}
+
+	var autodetectionV6Key string
+	var autodetectionV6Value any
+
+	if v, ok := annotations.Get(annotationAutodetectionV6FirstFound); ok {
+		if autodetectionV6Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v6 annotations found: %s", annotationAutodetectionV6FirstFound)
+		}
+		autodetectionV6Key = "firstFound"
+		autodetectionV6Value = v == "true"
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV6Kubernetes); ok {
+		if autodetectionV6Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v6 annotations found: %s", annotationAutodetectionV6Kubernetes)
+		}
+		autodetectionV6Key = "kubernetes"
+		autodetectionV6Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV6Interface); ok {
+		if autodetectionV6Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v6 annotations found: %s", annotationAutodetectionV6Interface)
+		}
+		autodetectionV6Key = "interface"
+		autodetectionV6Value = v
+	}
+	if v, ok := annotations.Get(annotationAutodetectionV6SkipInterface); ok {
+		if autodetectionV6Key != "" {
+			return config, fmt.Errorf("multiple autodetection-v6 annotations found: %s", annotationAutodetectionV6SkipInterface)
```

Comment:
this error is not _too_ useful, as it does not say what other annotations are set (therefore is not easy to take action).

the autodetectionValue can be a `any`